### PR TITLE
Extended exception message for ConfigException to warn in case of vcs…

### PR DESCRIPTION
**Pull request by leifole**
… type change

Not sure which branch to create pull request for, though. stable is more actual than master? Or autoproj-2.0?

### CODE REVIEW

**Comment by leifole** [comment]
And the string concatenation in to_s looses the indentation...

**Comment by goldhoorn** [comment]
I think the best way would be to mark the VCS type and URL in each autobuild package by creating a hidden file. This file can then be checked by autobuild on each update, if the information are differ throw a warning and ask the user for deletion. 

This should finally fix all VCS-change related problems (url-change-update/repro change etc.)

File: lib/autobuild/exceptions.rb [hunk 1]
```diff
@@ -46,6 +46,10 @@ def initialize(target = nil, phase = nil, options = Hash.new)
                 retry: false
             super(target, phase, options.merge(other_options))
         end
+
+        def to_s
+            super + "\nIf the VCS type changed, please delete your working copy first."
```
Comment (goldhoorn) [inline lib/autobuild/exceptions.rb]: The user has no information if someone has changes the VCS. See recommend workflow in general discussion.
Furthermore extend a generic error message with specific assumptions feels wrong.

**Comment by leifole** [comment]
Agreed, assuming a vcs change in every situation gives me a little bit of a headache, too. But working with hidden files feels even wronger to me. And what recommended workflow in general discussion are you talking about? Do you have a link?

**Comment by doudou** [comment]
> Agreed, assuming a vcs change in every situation gives me a little bit of a headache, too

The better option would be to create a InvalidCheckout exception and go through the importers to raise it instead of the generic ConfigError whenever they check for the checkout type.

**Comment by doudou** [comment]
> I think the best way would be to mark the VCS type and URL in each autobuild package by creating a hidden file. This file can then be checked by autobuild on each update, if the information are differ throw a warning and ask the user for deletion.

:-1: Ugly, not working for manually checked out packages that are "taken over" by autoproj.

> This should finally fix all VCS-change related problems (url-change-update/repro change etc.)

Which are not problems. Git handles all of these very nicely. SVN and other centralized VCS have to handle them manually (or raise the InvalidCheckout exception).

**Comment by doudou** [comment]
> Not sure which branch to create pull request for, though. stable is more actual than master? Or autoproj-2.0?

Please use autoproj-2.0. I am going to announce and merge it to master in the next days.

**Comment by doudou** [comment]
Closing because stale ... Re-open if you feel like starting to work on it again.

**Pull request closed without merging**